### PR TITLE
Queries

### PR DESCRIPTION
**Pull request by DarVoid**

### CODE REVIEW

File: auth.go [hunk 1]
```diff
@@ -14,5 +14,6 @@ func auth2() (ctx context.Context) {
 		&oauth2.Token{AccessToken: os.Getenv(Contexts[currentContext].Github.Token)},
 	)
 	client = oauth2.NewClient(ctx, ts)
+
```
Comment (DarVoid) [inline auth.go:17]:
```suggestion
```

File: gitter/gitter.go [hunk 2]
```diff
@@ -7,8 +7,7 @@ import (
 	"github.com/machinebox/graphql"
 )
 
-func GetPullRequests(url string, data *PrResponse, token string, ctx context.Context) {
-
+func GetPullRequests(url string, data *PrResponse, token string, ctx context.Context) error {
 	req := graphql.NewRequest(
 		`query fetchPRs {
```
Comment (DarVoid) [inline gitter/gitter.go:12]: we should prolly give the personal queries the same treatment we are giving the labeled queries and repo queries

Comment (DarVoid) [reply gitter/gitter.go:12]: (template from file instead of hardcoded here)

File: gitter/gitter.go [hunk 3]
```diff
@@ -7,8 +7,7 @@ import (
 	"github.com/machinebox/graphql"
 )
 
-func GetPullRequests(url string, data *PrResponse, token string, ctx context.Context) {
```
Comment (DarVoid) [inline gitter/gitter.go:10]: also, return pull requests instead of sending a pointer xD

File: gitter/gitter.go [hunk 4]
```diff
@@ -66,13 +65,17 @@ func GetPullRequests(url string, data *PrResponse, token string, ctx context.Con
...
+		return fmt.Errorf("error requesting PRs: %w", err)
 	}
+
 	fmt.Println(data)
+
+	return nil
 }
 
-func ApprovePullRequest(url string, token string, ctx context.Context, id string, body string) { //TODO: fix the graphql injection xD
+func ApprovePullRequest(url string, token string, ctx context.Context, id string, body string) error { // TODO: fix the graphql injection xD
```
Comment (DarVoid) [inline gitter/gitter.go:78]: I also wanna look into whether there is a mutation to merge/ enable auto merge from th graphql api

File: gitter/gitter.go [hunk 5]
```diff
@@ -109,12 +115,12 @@ func MergePR(url string, data *PrResponse, token string, ctx context.Context) {
 	panic("NOT IMPLEMENTED")
 }
 
-// Gargabe start
+// Gargabe start.
```
Comment (DarVoid) [inline gitter/gitter.go:118]: extract this onto self container module for tracking personal prs

File: main.go [hunk 6]
```diff
@@ -59,7 +71,41 @@ func pushPR(pr gitter.PullRequest) {
...
+			gqlApi := getGraphQLApi()
+			err := gitter.ApprovePullRequest(gqlApi, token, ctx, pr.Id, "LGTM! 🚀")
+			if err != nil {
+				ehp(fmt.Errorf("error while auto-replying LGTM: %w", err))
+			}
+		},
+	})
+}
+
+func pushTrackedPR(pr gitter.PullRequest) {
```
Comment (DarVoid) [inline main.go:96]: this will prolly go away once we have the more generic approach

File: gitter/gitter.go (hunk 2 above)
Comment (tobyselway) [reply gitter/gitter.go:12]: :ok_hand:

File: gitter/gitter.go (hunk 3 above)
Comment (tobyselway) [reply gitter/gitter.go:10]: yes please :pray:

File: gitter/gitter.go (hunk 5 above)
Comment (tobyselway) [reply gitter/gitter.go:118]: next to the other "queries"?

File: gitter/gitter.go [hunk 7]
```diff
@@ -163,6 +168,11 @@ func (pr pullRequest) transform() PullRequest {
 		Mergeable:      pr.Mergeable,
 	}
 }
+
+type Displayable interface {
+	Extract() []PullRequest
+}
```
[the remark this answers is not in the document]
Comment (tobyselway) [reply gitter/gitter.go:174]: Perhaps move it closer to the UI code. Could be interesting to move the toolbar icon UI logic into its own package, and abstract it from PRs, so it only renders `Displayable`s

File: gitter/trackedPRs.go [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+type Resposta struct {
+	data map[string]TrackedPullRequest
+}
+
+type TrackedPullRequest struct {
+	label struct {
+		pullRequests struct {
+			edges []struct {
+				node struct {
+					id             string `yaml:"id"`
```
[the remark this answers is not in the document]
Comment (tobyselway) [reply gitter/trackedPRs.go:19]: agreed :vomiting_face:

File: main.go (hunk 6 above)
Comment (tobyselway) [reply main.go:96]:
```go
type Renderer interface {
    Render(elem Displayable)
}

type ToolbarRenderer struct {} // Implements Renderer
```

File: gitter/gitter.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (DarVoid) [reply gitter/gitter.go]: yeah, that'll be solved once we extract the ui logic onto a separate module, prolly in another PR

**Review by tobyselway — APPROVED** [review]
LPOTM (Looks Probably Okay To Me) :sweat_smile:

**Pull request merged**